sql/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java

```diff
@@ -108,6 +109,22 @@ public void testUnsupportedExpressionCurrentDate() throws Exception {
     }
 
     @Test
+    public void testQuotedSubscriptExpression() throws Exception {
+        ExpressionAnalyzer expressionAnalyzer = new ExpressionAnalyzer(mockedAnalysisMetaData, emptyParameterContext, new FullQualifedNameFieldProvider(dummySources));
+        ExpressionAnalysisContext expressionAnalysisContext = new ExpressionAnalysisContext();
+
+        Field field1 = (Field) expressionAnalyzer.convert(SqlParser.createExpression("obj['x']"), expressionAnalysisContext);
+        Field field2 = (Field) expressionAnalyzer.convert(SqlParser.createExpression("\"obj['x']\""), expressionAnalysisContext);
+        assertEquals(field1, field2);
+
+        assertFalse(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj"))));
+        assertFalse(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj.x"))));
+        assertFalse(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj[x][y]"))));
+        assertTrue(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj['x']"))));
+        assertTrue(ExpressionAnalyzer.isQuotedSubscript(new QualifiedNameReference(new QualifiedName("obj['x']['y']"))));
```

Comment:
pls add a test validating a subscript expr where quoting is really necessary like e.g. "myObj['x']"